pkg/api/types.go

```diff
@@ -1101,6 +1102,11 @@ func (m *MasterProfile) IsCustomVNET() bool {
 	return len(m.VnetSubnetID) > 0
 }
 
+// HasVMSSNodePublicIP returns true if EnableVMSSNodePublicIP flag is true.
+func (a *AgentPoolProfile) HasVMSSNodePublicIP() bool {
+	return a.EnableVMSSNodePublicIP != nil && *a.EnableVMSSNodePublicIP
```

Comment:
no need for this extra function, you can use the built-in `to.Bool()` func wherever this function is used instead:

`to.Bool(a.EnableVMSSNodePublicIP)`